### PR TITLE
chore(deps): update rollup dependencies

### PR DESCRIPTION
**Pull request by JustinBeckwith**
This includes:
- Moving `rollup` to a dev dependency
- Unpinning rollup dependencies and using a `^` specifier (check me here, not sure why it was that way)
- Updating all the rollup dependency versions

These were causing issues when trying to move to `npm`.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -78,6 +77,7 @@
     "lint-staged": "^13.1.0",
     "lodash.camelcase": "^4.3.0",
     "prettier": "^3.2.5",
+    "rollup": "^4.14.1",
```
Comment (matthova) [inline package.json:80]: not sure why these were pinned, but lgtm, pending a quick SDK playground smoke-test

Comment (JustinBeckwith) [reply package.json:80]: did a smoke test, appears to be good :)

Comment (JustinBeckwith) [reply package.json:80]: thought waiting for an actual stamp

**Review by afgiel — APPROVED** [review]
lgtm

**Pull request merged**